data/basics/address.go

```diff
@@ -29,8 +29,8 @@ type (
 	Address crypto.Digest
 )
 
-// ChecksumAddress is a representation of the short address with a checksum
-type ChecksumAddress struct {
+// checksumAddress is a representation of the short address with a checksum
```

Comment:
is there anything still referencing `checksumAddress` or can we just delete it?

Reply:
It's used internally by address.go and that's it. I considered moving all of the functionality into Address struct, but the breakdown of responsibilities between the two types is both reasonable and eloquent.  With the private type checksumAddress being responsible for calculating, validating checksums etc.